### PR TITLE
Optimizer

### PR DESCRIPTION
**Pull request by maciekdragula**

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -104,30 +107,31 @@ def calc(self, inputs):
...
         # If the early stopping flag is off then run the training for the number of epochs and save the model
         # every (period) epochs
         else:
-            c_p = keras.callbacks.ModelCheckpoint("%sVAE-{epoch:02d}.h5" % self.checkpoint_dir, monitor="val_loss",
-                                                  verbose=0, save_best_only=False, save_weights_only=False, mode="auto",
-                                                  period=100)  # the model will be saved every 100 epochs
+            c_p = ModelCheckpoint(f"{self.checkpoint_dir}VAE-{{epoch:02d}}.h5", monitor="val_loss",
+                                  verbose=0, save_best_only=False, save_weights_only=False,
+                                  mode="min",
+                                  save_freq=100)  # the model will be saved every 100 epochs
```
Comment (DalilaSalamani) [inline model.py]: Please add the value of "save_freq" in ModelCheckpoint in constants

File: README.md [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 - train: performs model training.
-- generate: generate showers using a saved VAE model. 
-- observables: defines a set of shower observables. 
-- validate: creates validation plots using shower observables. 
+- generate: generate showers using a saved VAE model.
+- observables: defines a set of shower observables.
+- validate: creates validation plots using shower observables.
 - convert: defines the conversion function to and ONNX file.
+- optimizer: defines the Optimizer class.
+- optimize: performs hyperparameters optimization.
```
Comment (DalilaSalamani) [inline README.md:14]: Can you also add here how to run this script?

Comment (maciekdragula) [reply README.md:14]: Done.

File: model.py (hunk 1 above)
Comment (maciekdragula) [reply model.py]: Done.

**Review by DalilaSalamani — APPROVED** [review]

**Pull request merged**